core/lib/spree/testing_support/caching.rb

```diff
@@ -24,6 +28,11 @@ def clear_cache_events
       @cache_write_events ||= []
       @cache_write_events << details
     end
+
+    ActiveSupport::Notifications.subscribe("render_collection.action_view") do |_event, _start_time, _finish_time, _, details|
```

Comment:
Sorry, I quite don't get this. Can you please explain why we should treat `render_collection.action_view` as a cache write event?

Reply:
Rails fires this event whenever a collection is rendered, and it also includes information about how many elements were fetched from the cache and how many rendered from scratch, that's why I'm using it to count how many were fetched from the cache in the method above.

Reply:
Ok, maybe I get it better. Still not sure why putting those events into the `@cache_write_events` array since they are not "writes". Maybe we could create a new object and put into it just the relevant information that we need? Something like:

```
ActiveSupport::Notifications.subscribe("render_collection.action_view") do |_event, _start_time, _finish_time, _, details|
  @cached_collection_items_count ||= 0
  @cached_collection_items_count += details[: cache_hits]
```

What do you think?